spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/JdbcSessionConfiguration.java

```diff
@@ -56,13 +58,28 @@ public JdbcSessionDatabaseInitializer jdbcSessionDatabaseInitializer(
 	public static class SpringBootJdbcHttpSessionConfiguration
 			extends JdbcHttpSessionConfiguration {
 
+		private SessionProperties sessionProperties;
+
 		@Autowired
 		public void customize(SessionProperties sessionProperties) {
-			Integer timeout = sessionProperties.getTimeout();
+			this.sessionProperties = sessionProperties;
+			Integer timeout = this.sessionProperties.getTimeout();
 			if (timeout != null) {
 				setMaxInactiveIntervalInSeconds(timeout);
 			}
-			setTableName(sessionProperties.getJdbc().getTableName());
+			setTableName(this.sessionProperties.getJdbc().getTableName());
+		}
+
+		@PostConstruct
+		public void checkInitializerConfig() {
+			Jdbc jdbcProperties = this.sessionProperties.getJdbc();
+			if (jdbcProperties.getInitializer().isEnabled()) {
+				if (Jdbc.DEFAULT_SCHEMA_LOCATION.equals(jdbcProperties.getSchema()) &&
+						!Jdbc.DEFAULT_TABLE_NAME.equals(jdbcProperties.getTableName())) {
```

Comment:
I see no reason to move that business in `@PostConstruct`. You're not using the state of the (Configuration) class but solely the state of `SessionProperties` so you could just as well move that in `SessionProperties` itself if that's an invalid state (more on this later).